omigami/feature_selector.py

```diff
@@ -453,7 +441,61 @@ def _make_progress_bar():
             return progressbar.NullBar
         return progressbar.ProgressBar
 
-    def export_average_feature_ranks(self, output_path: str):
-        # make df
-        # save to path as csv
-        pass
+    def export_average_feature_ranks(
+        self,
+        output_path: str,
+        feature_names: List[str] = None,
+        exclude_unused_features: bool = True,
+    ) -> pd.DataFrame:
+        """
+        Creates and saves dataframe from the feature selection results. This dataframe contains
+        the columns 'mid', 'mid', and 'max', the indices are the features and the values
+        are the average rank across repetitions and outer loops.
+
+        Parameters
+        ----------
+        output_path: str
+            Path where to save the csv dataframe
+        feature_names: List[str]
+            The name of every feature, by default None
+        exclude_unused_features: bool
+            Whether to remove the features that werent selected or not.
+
+        Returns
+        -------
+        pd.DataFrame:
+            Pandas dataframe containing the average feature ranks
+
+        """
+        ranks_df = self.get_average_ranks_df(feature_names, exclude_unused_features)
+
+        ranks_df.to_csv(output_path)
+        return ranks_df
+
+    def get_average_ranks_df(
+        self,
+        feature_names: Union[None, List[str]],
```

Comment:
probably it should be kwarg too, since in principle could be used independently from the export method.

Reply:
:+1: